### PR TITLE
05 07 19 food create - TROUBLESHOOTING

### PR DESCRIPTION
**Pull request by Mackenzie-Frey**
### Create foods#create API endpoint
This endpoint is functional, but the test file is not running and further tests need to be made.

food_create_spec.js is currently not running. `Line 32` is setup for a status code of `20` to ensure the test fails. The only test that is failing is the one for the `:id` (not related)
@nagerz Any thoughts as to why the file isn't running?

image

I also tried changing the test syntax to match the other tests (and not use async testing)... but same issue, the file is not running (line 47 has the expectation as a `20` error intentionally, so that the test fails, but this is not occurring).
image

Failing test that occurs with both test syntaxes (the failing test is an unrelated test)
image

### User Story
#7 

Functional on Postman.
image

### CODE REVIEW

File: routes/api/v1/foods.js [hunk 1]
```diff
@@ -37,4 +34,42 @@ router.get("/:id", function(req, res, next) {
...
+    } else {
+      res.status(200).send(JSON.stringify(parsedFood(food[0])))
+    }
+  })
+  .catch(error => {
+    res.status(500).send({ error })
+  })
+})
+
+function parsedFood(food) {
```
Comment (nagerz) [inline routes/api/v1/foods.js:80]: This looks nice. Pretty clean so probably not worth refactoring into a serializer or something.

File: specs/endpoints/food_create_spec.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    test('it should create a food successfully', () => {
+      const newFood =  {
+        name: "Pringles",
+        calories: 27
+      }
+      return request(app).post("/api/v1/foods").then(response => {
+        expect(response.status).toBe(20)
+      })
+    })
+
```
Comment (nagerz) [inline specs/endpoints/food_create_spec.js]: It looks almost all right. Believe you need to send the newFood int he body. So try this for line 46:
```return request(app).post("/api/v1/foods").send(newFood).then(response => { ```

Comment (nagerz) [reply specs/endpoints/food_create_spec.js]: Alternatively/additionally, can try this syntax for async/await:
```
test('it should return a food object', async () => {
      await request(app).get("/api/v1/foods/1").then(async response => {
        await expect(response.body.id).toBe(1),
        await expect(response.body.name).toBe('Cheetos'),
        await expect(response.body.calories).toBe(30)
      });
    });
```

**Review by nagerz — COMMENTED** [review]
Overall looks real good and close. Just some test alterations and additions.

File: specs/endpoints/food_create_spec.js (hunk 2 above)
Comment (Mackenzie-Frey) [reply specs/endpoints/food_create_spec.js]: I'm not seeing what's different between the code you have for line 48 and the code I have. Could be a version thing... ?

**Comment by Mackenzie-Frey** [comment]
@nagerz Are the tests running on your machine? They are not on mine.

File: specs/endpoints/food_create_spec.js (hunk 2 above)
Comment (nagerz) [reply specs/endpoints/food_create_spec.js]: the "send(NewFood)". You need to send your request body (what you created) somehow.

File: specs/endpoints/food_create_spec.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+var shell = require('shelljs');
```
Comment (nagerz) [inline specs/endpoints/food_create_spec.js]: Typo in the filename preventing it from running. Change the filename to: 
```
food_create.spec.js
```

**Pull request closed without merging**